operator/controllers/lifecycle/keptnworkloadinstance/reconcile_deploymentstate.go

```diff
@@ -51,6 +52,13 @@ func (r *KeptnWorkloadInstanceReconciler) isReplicaSetRunning(ctx context.Contex
 	if err != nil {
 		return false, err
 	}
+
```

Comment:
can you add a unit test that covers this? I think if this one is taken as a reference, this should be possible with relatively low effort: https://github.com/keptn/lifecycle-toolkit/blob/419663aadc37a6310976cfe19a09397ba6abde5a/operator/controllers/lifecycle/keptnworkloadinstance/controller_test.go#L173

Reply:
I think we should do this either in another pr or leave it (currently it behaves the same way as for deployments and this is handled via replicasets)